core/src/main/scala/org/http4s/headers/Content-Length.scala

```diff
@@ -12,18 +12,24 @@ import org.http4s.util.Writer
   *
   * @param length the length; throws an `IllegalArgumentException` if negative
   */
-final case class `Content-Length`(length: Long) extends Header.Parsed {
-  require(length >= 0L)
-
+sealed abstract case class `Content-Length`(length: Long) extends Header.Parsed {
```

Comment:
I liked the `sealed abstract case class` because it's the only way to reclaim `apply` and suppress `copy`.  But yeah, I've used the `*Impl` name as the lone implementation in a few other places and generally like it.